src/auth.rs

```diff
@@ -4,22 +4,41 @@ use actix_web::{error, Error};
 /// Checks the authorization header for the specified token.
 ///
 /// `Authorization: (type) <token>`
-pub fn check(host: &str, headers: &HeaderMap, token: Option<String>) -> Result<(), Error> {
-    if let Some(token) = token {
-        if !token.is_empty() {
-            let auth_header = headers
-                .get(AUTHORIZATION)
-                .map(|v| v.to_str().unwrap_or_default())
-                .map(|v| v.split_whitespace().last().unwrap_or_default());
-            if auth_header.unwrap_or_default() != token {
-                log::warn!(
-                    "authorization failure for {} (header: {})",
-                    host,
-                    auth_header.unwrap_or("none"),
-                );
-                return Err(error::ErrorUnauthorized("unauthorized"));
+pub fn check(
+    host: &str,
+    headers: &HeaderMap,
+    token: Option<String>,
+    tokens: Option<Vec<String>>,
```

Comment:
Can we pass a single argument here (only `tokens`) instead?

Reply:
This would be a lot easier after the token was removed from the code (release 1.0.0). Otherwise I have to merge token and tokens before passing them into the check function. I can look into it though. In theory this should be easy, but my lack of Rust knowledge makes this is a challenge:

I don't know how to pass a reference of config into a function.

e.g.

```rust
let config = Config::parse(&config_path).expect("failed to parse config");

let tokens = merge_tokens(&config);
```

What is the type of config? I don't know how to declare the function. 

```
fn merge_tokens(config: &what-to-put-here) -> Option<Vec<String>> 
```

But I will look into it. This gives me the opportunity to learn more about Rust.

Reply:
I think I figured it out. I'll work on it.

Reply:
Done. It was remarkably easy. I think I am starting to understand Rust a bit better.